media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt

```diff
@@ -0,0 +1,328 @@
+         * Currently, we only support tracks defined in [ArticObject.audioCommentary].
+         *
+         * @see [ExoPlayer.prepare]
+         * @see [Player.setPlayWhenReady]
+         */
+        class Play(val audioFile: ArticObject) : PlayBackAction()
+
+        /**
+         * Pause the current track.
+         *
+         * You can continue where you left off with [Resume].
+         *
+         * @see [Player.setPlayWhenReady]
+         */
+        class Pause : PlayBackAction()
+
+        /**
+         * Resume the current track.
+         *
+         * You can also resume (without resetting the stream) by sending a subsequent [Play]
+         * action backed by the same [ArticObject].
+         *
+         * @see [Player.setPlayWhenReady]
+         */
+        class Resume : PlayBackAction()
+
+        /**
+         * Stop playback, seek back to the start of the file, and move this service into the background.
+         *
+         * @see [Player.stop]
+         */
+        class Stop : PlayBackAction()
+
+        /**
+         * Move playback to a specific temporal position.
+         *
+         * @param time number of milliseconds from the start of the track
+         */
+        class Seek(val time: Long) : PlayBackAction()
+    }
+
+    /**
+     * Current 'state' of [the ExoPlayer][player].
+     *
+     * We enforce the following states:
+     * * [Playing]
+     * * [Paused]
+     * * [Stopped]
+     */
+    sealed class PlayBackState(val articAudioFile: ArticAudioFile) {
+        class Playing(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+        class Paused(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+        class Stopped(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+    }
+
+    private val binder: Binder = AudioPlayerServiceBinder()
+    private lateinit var playerNotificationManager: PlayerNotificationManager
+    var articObject: ArticObject? = null
+
+    private val audioControl: Subject<PlayBackAction> = BehaviorSubject.create()
+    val audioPlayBackStatus: Subject<PlayBackState> = BehaviorSubject.create()
+    val currentTrack: Subject<ArticAudioFile> = BehaviorSubject.create()
+
+    val disposeBag = DisposeBag()
+
+    /**
+     * Returns an intent including the details of currently playing audio file.
+     */
+    fun getIntent(): Intent {
+        val audioIntent = "edu.artic.audio".asDeepLinkIntent()
+        audioIntent.putExtra("artic_object", articObject)
+        return audioIntent
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        setUpNotificationManager()
+        player.addListener(object : Player.DefaultEventListener() {
+            override fun onPlayerStateChanged(playWhenReady: Boolean, playbackState: Int) {
+                val articAudioFile = articObject?.audioCommentary?.first()?.audioFile
+                articAudioFile?.let {
+                    if (playWhenReady && playbackState == Player.STATE_READY) {
```

Comment:
replace with `when()` just open intention dialog, and use "convert to when". much more compact and readable.

Reply:
(personal preference to use `when` statements)

Reply:
Updated